### DISCUSSION TITLE
Release Candidate v0.3.0-rc.0 is published!

### DISCUSSION THREAD

**Discussion by brainlid** [brainlid/langchain#139] [Announcements]
There are some breaking changes around how callbacks work. See the [CHANGELOG](https://github.com/brainlid/langchain/blob/main/CHANGELOG.md#v030-rc0-2024-06-05) for details and migration steps.

Another breaking change is with `ChatGoogleAI` where the `version` field was renamed to `api_version`.

With the callback changes, now we have the ability to get API response headers with rate limits information. We also get token usage information!

Please try it out and open issues with any problems you encounter. Thank you for using the library and contributing to our growing little band. :slightly_smiling_face:

**Comment by derekkraan** [comment]
I have been running 0.3.0-rc.0 for a couple weeks now and no issues to report. I am using the token_usage callback and it also seems to be working. I also checked the OpenAI billing dashboard to compare and it all seemed to match perfectly.

**Comment by tubedude** [comment]
@brainlid what do you think of expanding documentation in this direction https://github.com/brainlid/langchain/compare/main...tubedude:langchain:docs
In this specific case I wanted to highlight required and optional parameters.

What to think of an AI library without AI generated documention? :)

**Reply by brainlid** [reply]
Ha! I kinda like it. It feels like there's a few issues though. Like the documentation is indented by 2 spaces which it shouldn't be. 

While handy, my main hesitance in creating a lot of doc text is that I didn't write it but have to maintain it. :smile: But having docs is good too. Feeling a little conflicted.
